crates/biome_js_analyze/src/lint/correctness/use_hook_at_top_level.rs

```diff
@@ -231,6 +233,12 @@ fn is_nested_function_inside_component_or_hook(function: &AnyJsFunctionOrMethod)
     })
 }
 
+fn is_within_react_project(manifest: &Option<PackageJson>) -> bool {
+    manifest.as_ref().map_or(false, |package_json| {
+        package_json.dependencies.contains("react")
+    })
```

Comment:
I think we should return `true` by default if the manifest is not found. This ensures that the rule is usable even if we are not able to find the `package.json`.

```suggestion
    manifest.as_ref().map_or(true, |package_json| {
        package_json.dependencies.contains("react")
    })
```